### PR TITLE
Feature: Scroll Stripe checkout modal into view for Legacy + Multistep forms

### LINKED ISSUE #6788: Stripe Checkout modal should immediately be visible on all views and form lengths
## Details
When processing a donation with Stripe checkout modal, the credit card fields sometimes cannot be seen without user scrolling if the donation form is too long or when the user's is on a device with screen sizing similar to mobile views. This is problematic because users may not be aware that they need to scroll up to enter their credit card details, potentially leading to abandoned donations. 

## Expected Behavior
When users reach the Stripe modal during the donation process, the checkout process should allow users to see the credit card fields without requiring any scrolling, regardless of the length of the donation form or the user's screen size.

## Steps to Reproduce

- Create a form and set Stripe Checkout as a payment option.
- Begin the donation process on a mobile device, a long donation form, or with altered viewports on a desktop.
- Fill out the form until you reach the second to last step before payment.
- Click on the "Donate Now" button.
- Observe that the Stripe modal appears in the middle of the page and will sometimes require the user to scroll up to view it.

## Visuals
stripemodal2
stripemodal

https://drive.google.com/file/d/1tOJ4C1OdMbCCKhxU9fK9Ssw-lbek9AH-/view?usp=sharing

## Acceptance Criteria

-   [x] When the Stripe checkout modal is populated the user can immediately see the required credit card fields.

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves #6788

## Description

This PR adds functionality to scroll the Stripe checkout modal into view when the user clicks the purchase button. The checkout modal will now be visible to the user, regardless of their screen size. This fix currently only works for legacy & classic templates.

## Affects

Stripe checkout modal

## Testing Instructions

- Create a form and set Stripe Checkout as a payment option.
- Begin the donation process on a mobile device, a long donation form, or with altered viewports on a desktop.
- Fill out the form until you reach the second to last step before payment.
- Click on the "Donate Now" button.
- The Stripe modal should scroll into view.

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204508757292153

### CODE REVIEW

File: assets/src/js/frontend/give-stripe-elements.js [hunk 1]
```diff
@@ -308,15 +308,24 @@ class GiveStripeElements {
...
 		const cardholderName = formElement.querySelector( 'input[name="card_name"]' );
+        const stripeModalContent = document.querySelector('.give-stripe-checkout-modal-container')
+        const purchaseButton = document.querySelector('#give-purchase-button')
 		const completeCardElements = {};
 		let completeCardStatus = false;
 
+        // Scroll checkout modal container into view.
+        purchaseButton.addEventListener('click', function(){
+            stripeModalContent.scrollIntoView({ behavior: 'smooth' });
+        })
```
Comment (JasonTheAdams) [inline assets/src/js/frontend/give-stripe-elements.js:327]: Formatting looks a touch off. Is prettier formatting doing its thing?

**Review by JasonTheAdams — APPROVED** [review]
Looks good, @JoshuaHungDinh! This should work. I just notice a super minor formatting thing.

File: assets/src/js/frontend/give-stripe-elements.js (hunk 1 above)
Comment (JoshuaHungDinh) [reply assets/src/js/frontend/give-stripe-elements.js:327]: Yes, this was formatted by prettier. 

https://user-images.githubusercontent.com/75056371/235512220-307d420c-f962-4c44-b4d6-5bf9fed6f720.mov

**Comment by jonwaldstein** [comment]
@JoshuaHungDinh what's the status of this?

**Comment by JoshuaHungDinh** [comment]
@jonwaldstein So there was some back and forth on the PR because it would only scroll the stripe modals for Legacy and Multistep form. It would not work on classic. We are left with two options. Either merge a partial fix or close it out. Heres the last conversations. I think the verdict was a little unclear due to how much time had passed. What do you think? https://app.asana.com/0/1203193833018151/1204508757292153

**Comment by jonwaldstein** [comment]
@JoshuaHungDinh gotchya, I would go with partial fix and change the PR title/scope to state that it's just for legacy & multistep.  Classic template is still the least used one so at least this would be fixed for folks using the other templates and we can circle back if need be.

**Pull request merged**